### PR TITLE
chore: wrap error objects to include context (#10592)

### PR DESCRIPTION
**Pull request by aimanfatima**
Signed-off-by: Aiman Fatima <aimanfatimadl@gmail.com>
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by aimanfatima** [comment]
This PR is related to this issue -  https://github.com/argoproj/argo-cd/issues/10592

File: applicationset/services/repo_service.go [hunk 1]
```diff
@@ -85,12 +85,12 @@ func (a *argoCDService) GetDirectories(ctx context.Context, repoURL string, revi
 
 	gitRepoClient, err := git.NewClient(repo.Repo, repo.GetGitCreds(a.storecreds), repo.IsInsecure(), repo.IsLFSEnabled(), repo.Proxy)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Error in NewClient: %w", err)
```
Comment (crenshaw-dev) [inline applicationset/services/repo_service.go:88]:
```suggestion
		return nil, fmt.Errorf("error creating a new git client: %w", err)
```

File: applicationset/services/repo_service.go [hunk 2]
```diff
@@ -85,12 +85,12 @@ func (a *argoCDService) GetDirectories(ctx context.Context, repoURL string, revi
...
 	}
 
 	err = checkoutRepo(gitRepoClient, revision, a.submoduleEnabled)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Error while checking out repo: %w", err)
```
Comment (crenshaw-dev) [inline applicationset/services/repo_service.go:93]:
```suggestion
		return nil, fmt.Errorf("error while checking out repo: %w", err)
```

File: applicationset/services/repo_service.go [hunk 3]
```diff
@@ -99,7 +99,7 @@ func (a *argoCDService) GetDirectories(ctx context.Context, repoURL string, revi
 
 	if err := filepath.Walk(repoRoot, func(path string, info os.FileInfo, fnErr error) error {
 		if fnErr != nil {
-			return fnErr
+			return fmt.Errorf("Error walking the file tree: %w", fnErr)
```
Comment (crenshaw-dev) [inline applicationset/services/repo_service.go:102]:
```suggestion
			return fmt.Errorf("error walking the file tree: %w", fnErr)
```

File: applicationset/services/repo_service.go [hunk 4]
```diff
@@ -112,7 +112,7 @@ func (a *argoCDService) GetDirectories(ctx context.Context, repoURL string, revi
 
 		relativePath, err := filepath.Rel(repoRoot, path)
 		if err != nil {
-			return err
+			return fmt.Errorf("Error fetching the relative path: %w", err)
```
Comment (crenshaw-dev) [inline applicationset/services/repo_service.go:115]:
```suggestion
			return fmt.Errorf("error constructing relative repo path: %w", err)
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Thanks @aimanfatima! Just wording nitpicks. Golang also has a convention of lowercase first letters in error messages. I don't know why.  :-P

File: applicationset/services/repo_service.go (hunk 4 above)
Comment (aimanfatima) [reply applicationset/services/repo_service.go]: updated

**Review by crenshaw-dev — APPROVED** [review]
Thanks @aimanfatima!

**Pull request merged**